packages/opentelemetry-shim-opentracing/src/shim.ts

```diff
@@ -125,29 +150,39 @@ export class TracerShim extends opentracing.Tracer {
       getContextWithParent(options)
     );
 
+    let correlationContext: api.CorrelationContext = {};
+    if (options.childOf) {
+      if (options.childOf instanceof SpanShim) {
+        const shimContext = options.childOf.context() as SpanContextShim;
+        correlationContext = shimContext.getCorrelationContext();
+      }
+    }
+
     if (options.tags) {
       span.setAttributes(options.tags);
     }
 
-    return new SpanShim(this, span);
+    return new SpanShim(this, span, correlationContext);
   }
 
   _inject(
     spanContext: opentracing.SpanContext,
     format: string,
     carrier: unknown
   ): void {
-    const opentelemSpanContext: api.SpanContext = (spanContext as SpanContextShim).getSpanContext();
+    const oTelSpanContext: api.SpanContext = (spanContext as SpanContextShim).getSpanContext();
```

Comment:
you are casting here twice for `spanContext as SpanContextShim`. Perhaps you can use some variable cast it once and then use it in both places ?

Reply:
+1